lib/common/cmdline.c

```diff
@@ -144,6 +144,74 @@ pcmk__add_arg_group(GOptionContext *context, const char *name,
     // cppcheck-suppress memleak
 }
 
+static gchar *
+string_replace(gchar *str, const gchar *sub, const gchar *repl)
+{
+    /* This function just replaces all occurrences of a substring
+     * with some other string.  It doesn't handle cases like overlapping,
+     * so don't get clever with it.
+     *
+     * FIXME: When glib >= 2.68 is supported, we can get rid of this
+     * function and use g_string_replace instead.
+     */
+    gchar **split = g_strsplit(str, sub, 0);
+    gchar *retval = g_strjoinv(repl, split);
+
+    g_strfreev(split);
+    return retval;
+}
+
+gchar *
+pcmk__quote_cmdline(gchar **argv)
+{
+    GString *gs = NULL;
+    gchar *retval = NULL;
+
+    if (argv == NULL || argv[0] == NULL) {
+        return NULL;
+    }
+
+    gs = g_string_sized_new(100);
+
+    /* The first item is the program name. */
+    g_string_append(gs, argv[0]);
```

Comment:
This would be fine in all the cases we really care about, since we know the universe of possible command names, but there is the theoretical possibility that a user could install pacemaker on a path that includes a space or quote and then run the command using the full path (or that a user could rename or symlink a binary to a name with a space or quote, but that's even less realistic).

Bottom line I'm OK with this if you are, but maybe at least comment the issue

Reply:
I don't think it would cause any problems to just get rid of this line and run the program name through the for loop as well.

Incidentally, this made me look at pcmk__cmdline_preproc again.  There, we take the first thing in argv as the program name but still loop over argv entirely.  Unit tests for that function, however, leave the program name out.  I wonder if there's a bug hiding in there somewhere.